dockstore-webservice/src/main/java/io/dockstore/webservice/resources/WorkflowResource.java

```
@@ -1292,14 +1268,158 @@ public void unstarEntry(@ApiParam(hidden = true) @Auth User user,
+        // Find the entry
+        MutablePair<String, Entry> entryPair = toolDAO.findEntryById(entryId);
+
+        // Check if valid descriptor type
+        if (!Objects.equals(descriptorType, "cwl") && !Objects.equals(descriptorType, "wdl")) {
+            throw new CustomWebApplicationException(descriptorType + " is not a valid descriptor type. Only cwl and wdl are valid.", HttpStatus.SC_BAD_REQUEST);
+        }
+
+        // Check if the entry exists
+        if (entryPair.getValue() == null) {
+            throw new CustomWebApplicationException("No entry with the given ID exists.", HttpStatus.SC_BAD_REQUEST);
+        }
+
+        // Don't allow workflow stubs
+        if (Objects.equals(entryPair.getKey(), "workflow")) {
+            Workflow workflow = (Workflow) entryPair.getValue();
+            if (Objects.equals(workflow.getMode().name(), "STUB")) {
+                throw new CustomWebApplicationException("Checker workflows cannot be added to workflow stubs.", HttpStatus.SC_BAD_REQUEST);
+            }
+        }
+
+        // Ensure that the entry has no checker workflows already
+        if (entryPair.getValue().getCheckerWorkflow() != null) {
+            throw new CustomWebApplicationException("The given entry already has a checker workflow.", HttpStatus.SC_BAD_REQUEST);
+        }
+
+        // Checker workflow variables
+        String defaultTestParameterPath;
+        String organization;
+        String repository;
+        String sourceControl;
+        boolean isPublished;
+        String gitUrl;
+        Date lastUpdated;
+        String workflowName;
+
+        // Grab information if tool
+        if (Objects.equals(entryPair.getKey(), "tool")) {
+            // Get tool
+            Tool tool = (Tool)entryPair.getValue();
+
+            // Get default test parameter path and toolname
+            if (Objects.equals(descriptorType.toLowerCase(), "wdl")) {
```

Throw an UnsupportedOperationException or similar for the else rather than assuming all non-wdl workflows are CWL (e.g. NextFlow futureproofing). Also, we probably have an enum for this